crates/sui-core/src/authority/authority_per_epoch_store.rs

```diff
@@ -3866,6 +3771,299 @@ impl AuthorityPerEpochStore {
     }
 }
 
+#[derive(Default)]
+pub(crate) struct ConsensusCommitOutput {
+    // Consensus and reconfig state
+    consensus_messages_processed: BTreeSet<SequencedConsensusTransactionKey>,
+    end_of_publish: BTreeSet<AuthorityName>,
+    reconfig_state: Option<ReconfigState>,
+    consensus_commit_stats: Option<ExecutionIndicesWithStats>,
+    pending_execution: Vec<VerifiedExecutableTransaction>,
+
+    // transaction scheduling state
+    shared_object_versions: Option<(AssignedTxAndVersions, HashMap<ObjectID, SequenceNumber>)>,
+
+    deferred_txns: Vec<(DeferralKey, Vec<VerifiedSequencedConsensusTransaction>)>,
+    // deferred txns that have been loaded and can be removed
+    deleted_deferred_txns: BTreeSet<DeferralKey>,
+
+    // checkpoint state
+    user_signatures_for_checkpoints: Vec<(TransactionDigest, Vec<GenericSignature>)>,
+    pending_checkpoints: Vec<PendingCheckpointV2>,
+
+    // random beacon state
+    next_randomness_round: Option<(RandomnessRound, TimestampMs)>,
+
+    dkg_confirmations: BTreeMap<PartyId, VersionedDkgConfirmation>,
+    dkg_processed_messages: BTreeMap<PartyId, VersionedProcessedMessage>,
+    dkg_used_message: Option<VersionedUsedProcessedMessages>,
+    dkg_output: Option<dkg::Output<PkG, EncG>>,
+
+    // jwk state
+    pending_jwks: BTreeSet<(AuthorityName, JwkId, JWK)>,
+    active_jwks: BTreeSet<(u64, (JwkId, JWK))>,
+}
+
+impl ConsensusCommitOutput {
+    pub fn new() -> Self {
+        Default::default()
+    }
+
+    fn insert_end_of_publish(&mut self, authority: AuthorityName) {
+        self.end_of_publish.insert(authority);
+    }
+
+    fn insert_pending_execution(&mut self, transactions: &[VerifiedExecutableTransaction]) {
+        self.pending_execution.reserve(transactions.len());
+        self.pending_execution.extend_from_slice(transactions);
+    }
+
+    fn insert_user_signatures_for_checkpoints(
+        &mut self,
+        transactions: &[VerifiedExecutableTransaction],
+    ) {
+        self.user_signatures_for_checkpoints
```

Comment:
could this be simplified (no separate call to `reserve`) by using `extend` and an iter on `transactions`?

Reply:
sure. extend supposedly uses size_hint() and will reserve, although I with rust was clearer about this in the docs